safe_app/src/ffi/access_container.rs

```diff
@@ -64,7 +64,10 @@ pub unsafe extern "C" fn access_container_fetch(
                 .get_access_info(client)
                 .and_then(move |containers| {
                     let ffi_containers = containers_into_vec(
-                        containers.into_iter().map(|(key, (_, value))| (key, value)),
+                        containers
+                            .1
```

Comment:
That's because we have both `(AuthTokens, Containers)` living inside the `AccessContainer` now(as a tuple). Hence, the `.1` wherever we want to point to containers

Reply:
aha, didn't know about that notation.

For clarity, maybe we can setup `let containers = auth_data_struct.1` and use that all over?

Right now its confusing, cos we're getting `.1` from `containers` anyway, so i think we need to rename that data struct var to something else, and then also declare `containers` separately so it's clear what we're getting.

Thoughts?

Reply:
is it possible to use destructuring here for even more clarity? not sure how that goes in rust:

`let (_, containers) = that_struct` ?

Reply:
>is it possible to use destructuring here for even more clarity?

Yup, that is the right way to do this. I'll get in this refactor as well.